### ISSUE TITLE
Auto-completion causes browser to crash within Razor

### ISSUE THREAD

**Issue by sihoran** [Microsoft/monaco-editor#464]
monaco-editor version: 0.8.3
Browser: Chrome 58, Firefox 53, IE 11, IE Edge
OS: Windows 10
Steps or JS usage snippet reproducing the issue:

If you have the following code example in the editor

```
@{ 
//start adding an html element under this line to freeze the browser window/tab


}

<script>1+1;</script>
```

then start to type a HTML element within the {} the browser crashes.

A live example can be seen here - https://plnkr.co/edit/5C9R0bh8oXnrQL6p0tuU?p=preview

**Comment by acip** [comment]
@sihoran This was discussed here: https://github.com/Microsoft/monaco-editor/issues/388 and those changes fix the editor. They just need to be pulled from vscode to Monaco.

@alexandrudima When can we expect these changes to be brought to Monaco? In the meanwhile, could you reference the documentation for generating Monaco from vscode code?

**Comment by alexandrudima** [comment]
I've just checked that the fix for #388 also fixes this issue.
